base/abstractarray.jl

```diff
@@ -38,18 +38,12 @@ function first(itr)
 end
 last(a) = a[end]
 
-function stride(a::AbstractArray, i::Integer)
-    if i > ndims(a)
-        return length(a)
-    end
-    s = 1
-    for n=1:(i-1)
-        s *= size(a, n)
-    end
-    return s
-end
+strides{T}(a::AbstractArray{T,0}) = ()
+strides(a::AbstractArray) = _strides(a,(1,))
+_strides{_,N}(a::AbstractArray{_,N}, s::NTuple{N,Int}) = s
+_strides{_,N,M}(a::AbstractArray{_,N}, s::NTuple{M,Int}) = _strides(a, (s..., s[M]*size(a, M)))
 
-strides(a::AbstractArray) = ntuple(i->stride(a,i), ndims(a))::Dims
+stride(a::AbstractArray, d) = d <= 1 ? 1 : size(a, d-1)*stride(a, d-1)
```

Comment:
For me, the first one evaluates in the same time, the second one in about 0.06 seconds, so about 20 % slower. The fastest one, however, seems to be the following definitition
```julia
mystride2(a::AbstractArray, d) = d < 1 ? 1 : d > ndims(a) ? length(a) : strides(a)[d]
```
where `strides` is definitely already the one from this PR.